File: VectorNotes.DomainModel/NoteCollection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace VectorNotes.DomainModel
{
    public class NoteCollection
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int OwnerId { get; set; }
        public User? Owner { get; set; }

        public IList<Note> Notes { get; set; }

        public NoteCollection()
        {
            Name = string.Empty;
            Notes = [];
        }
    }
}

File: vectornotes.client/src/app/model/note-collection.ts
import { NotePreview } from "./note-preview";

export class NoteCollection {
  id?: number;
  name?: string;
  notes?: NotePreview[];

  constructor(source: NoteCollection | undefined = undefined) {

    if (!source) return;

    this.id = source.id;
    this.name = source.name;
    this.notes = source.notes ? [...source.notes] : [];
  }
}
